### ISSUE TITLE
CCM GDBM issue with gdbm 1.8.0-37 update

### ISSUE THREAD

**Issue by stdweird** [quattor/CCM#40]
a GDBM update broke CCM with GDBM_File backend (http://lists.centos.org/pipermail/centos-announce/2015-January/020856.html)

error message is like 
```
gdbm store returned -1, errno 0, key "/system/network/interfaces/eth1/route/7/gateway" at /usr/lib/perl/EDG/WP4/CCM/DB.pm line 98.
```

if you are using yum-spma
```pan
"/software/packages" = pkg_repl('gdbm*','1.8.0-36.el6', PKG_ARCH_DEFAULT);
```

(or switch format)

if you already have the issue on nodes:
* lock the version / switch format, make new profiles
* on the node
```shell
yum -y downgrade gdbm-1.8.0-36.el6.x86_64
ccm-fetch --force
```

**Comment by jrha** [comment]
So... should we change the default to something else?

**Comment by stdweird** [comment]
lets wait for the patch. something tells me it will pretty simple, looking at the changelog of the rpm. we can discuss about a new default later.

**Comment by stdweird** [comment]
reprodce with minimal code
```perl
#!/usr/bin/perl

use GDBM_File;

my $file = "/tmp/x";
my %out;
tie(%out, 'GDBM_File', "$file.db", &GDBM_WRCREAT, 0644)
    or print "can't tie $prefix DB: $!";

print "Adding data to tie\n";
$out{1}=2;

print "Going to untie\n";

untie(%out) or print "can't untie $prefix DB: $!";

```

gives
```
# rm -f /tmp/x.db ;./t.pl
Adding data to tie
gdbm store returned -1, errno 0, key "1" at ./t.pl line 11.
```

**Comment by stdweird** [comment]
should be fixed with https://rhn.redhat.com/errata/RHBA-2015-0089.html
waiting till it hits our repos to test it.

**Comment by jrha** [comment]
@stdweird any update?

**Comment by stdweird** [comment]
this is fixed with `gdbm-1.8.0-38.el6.x86_64` and available. 

If the version is in your repos (be careful with snapshots!), you can remove the whole `pkg_repl('gdbm*','1.8.0-36.el6', PKG_ARCH_DEFAULT)` again (gdbm will be pulled in as a dependency).

to verify if gdbm is in your repos, run
```bash
repoquery gdbm
```
(by default, `repoquery` does not honour yum version locks, so no need to make any changes to test this).
